index.bs

```diff
@@ -632,19 +632,281 @@ Issue(12): Propose control protocol for Remote
 Playback API.
 
 
-Security and Privacy {#security}
-================================
-
-Issue(13): Describe security architecture.
+Security and Privacy {#security-privacy}
+====================
 
-Data to be secured {#security-data}
+The Open Screen Protocol allows two networked user agents to discover each other
+and exchange user and application data.  As such, its security and privacy
+considerations should be closely examined.  We first evaluate the protocol
+itself using the W3C [[SECURITY-PRIVACY-QUESTIONNAIRE|Security and Privacy
+Questionnaire]].  We then examine whether the security and privacy guidelines
+recommended by the [[PRESENTATION-API|Presentation API]] and the
+[[REMOTE-PLAYBACK|Remote Playback API]] are met.  Finally we discuss recommended
+mitigations that implementations can use to meet these security and privacy
+requirements.
+
+Threat Models {#threat-models}
+--------------------------------
+
+### Passive Network Attackers ### {#passive-network-attackers}
+
+The Open Screen Protocol should assume that all parties that are connected to
+the same LAN, either through a wired connection or through WiFi, are able to
+observe all data flowing between Open Screen controllers and receivers.
+
+These parties will be able collect any data exposed through unencrypted
+messages, such as mDNS records and the QUIC handshake.
+
+These parties may attempt to learn cryptographic parameters by observing data
+flows on the QUIC connection, or by observing cryptographic timing.
+
+### Active Network Attackers ### {#active-network-attackers}
+
+In addition, all parties connected to the LAN will be able to manipulate data
+exchanged between controllers and receivers.  They will be able to inject
+traffic into existing QUIC connections and attempt to inititate new QUIC
+connections.  These abilities can be used to attempt the following:
+
+*   Impersonate a new receiver or one already trusted by the user, in an attempt
+    to convince the user to authenticate to it.
+*   Connect to a receiver and query its capabilities.
```

Comment:
Are we going to do capability-query before or after authentication?  We probably want some capabilities to be known before authentication.  But perhaps not all.  I can't think of a capability that you wouldn't want known.  Can you?

Reply:
I would prefer post-authentication by default.  The capabilities expose some facts about the devices in your vicinity to third parties, which could collect/mine that data for analytics or advertising purposes.   Also, we don't want the capabilities to be modified by a MITM (see the PII section).

If there's a use case for exposing them pre-authentication, we can explore that tradeoff.